src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringFromArray = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && typeof this[i] !== 'undefined') {
+        stringFromArray += `${this[i]}`;
+
+        return stringFromArray;
+      } else if (i === this.length - 1) {
+        return stringFromArray;
+      }
+
+      if (typeof this[i] !== 'undefined' && typeof this[i] !== 'object') {
```

Comment:
We need to check for nulls here, not objects. Your join works incorrectly because of this - `[{}, {}, 1].join2(',')` returns different result compared to real `.join`.
```suggestion
      if (typeof this[i] !== 'undefined' && this[i] !== null) {
```